contrib/filter-repo-demos/git_lint_history.py

```diff
@@ -149,24 +147,37 @@ def lint_non_binary_blobs(blob, metadata):
       blob.data = f.read()
     os.remove(filename)
 
-if lint_args.filenames_important and not lint_args.relevant:
-  lint_args.relevant = 'return True'
-if lint_args.relevant:
-  body = lint_args.relevant
-  exec('def is_relevant(filename):\n  '+'\n  '.join(body.splitlines()),
-       globals())
-  lint_args.filenames_important = True
-args = fr.FilteringOptions.default_options()
-args.force = True
-if lint_args.filenames_important:
-  tmpdir = tempfile.mkdtemp().encode()
-  cat_file_process = subprocess.Popen(['git', 'cat-file', '--batch'],
-                                      stdin = subprocess.PIPE,
-                                      stdout = subprocess.PIPE)
-  filter = fr.RepoFilter(args, commit_callback=lint_with_real_filenames)
-  filter.run()
-  cat_file_process.stdin.close()
-  cat_file_process.wait()
-else:
-  filter = fr.RepoFilter(args, blob_callback=lint_non_binary_blobs)
-  filter.run()
+def main(argv1=None):
+  global is_relevant, lint_args, tmpdir, cat_file_process, filter
+
+  if isinstance(argv1, str):
+    argv1 = shlex.split(argv1)
```

Comment:
Where/how is this meant to be used?  It's a new usage that doesn't appear anywhere in current usage, so what's the motivating usecase(s) and what are you trying to accomplish?  Again, this may belong in a separate commit with its own explanation.